python/pyspark/sql/connect/client/core.py

```diff
@@ -1525,9 +1525,15 @@ def _display_server_stack_trace(self) -> bool:
         from pyspark.sql.connect.conf import RuntimeConf
 
         conf = RuntimeConf(self)
-        if conf.get("spark.sql.connect.serverStacktrace.enabled") == "true":
+        try:
```

Comment:
I piggiy back a fix since I am here. This is tested together by the test added in `test_reattach`. cc @heyihong